cocos/platform/CCFileUtils.h

```diff
@@ -41,6 +42,66 @@ NS_CC_BEGIN
  * @{
  */
 
+
+class ResizableBuffer {
+public:
+    virtual ~ResizableBuffer() {}
+    virtual void resize(size_t size) = 0;
+    virtual void* buffer() const = 0;
+};
+
+template<typename T>
+class ResizableBufferAdapter { };
+
+
+template<typename C, typename T, typename A>
+class ResizableBufferAdapter< std::basic_string<C, T, A> > : public ResizableBuffer {
+    typedef std::basic_string<C, T, A> BufferType;
+    BufferType* _buffer;
+public:
+    explicit ResizableBufferAdapter(BufferType* buffer) : _buffer(buffer) {}
+    virtual void resize(size_t size) override {
+        _buffer->resize((size + sizeof(C) - 1) / sizeof(C));
```

Comment:
[`std::basic_string<>::size()`](http://en.cppreference.com/w/cpp/string/basic_string/size) and [`std::basic_string<>::resize()`](http://en.cppreference.com/w/cpp/string/basic_string/resize) is talking about the number of elements (yes the name make users think it's bytes size).